modules/core/src/lib/attribute/attribute-transition-utils.ts

```diff
@@ -167,21 +166,26 @@ export function padBuffer({
     ? (i, chunk) => getData(toData, chunk)
     : (i, chunk) => getData(toData.subarray(i + byteOffset, i + byteOffset + size), chunk);
 
-  const bufferWithAccessor = buffer as BufferWithAccessor;
-  const source = bufferWithAccessor.getData({length: fromLength});
-  const data = new Float32Array(toLength);
+  // TODO(donmccurdy): Might need a helper function here.
+  const sourceData = buffer.getData();
```

Comment:
What has happened to `fromLength`?

Reply:
Short answer: Oversight, and I've restored it for now, thanks!

Long answer: I think this is being used in combination with dynamic resizing of buffers, which is something we're going to want to avoid in v9. I'm also not sure whether we're keeping a synchronous method for reading buffers, since that doesn't appear to be supported in WebGPU, but switching to an async method here is a larger change than I wanted to try in this PR.